src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -943,43 +971,98 @@ protected override void ValidateElement(object element)
                         Metadata.ValidateNotNullFailure);
             }
 
-            var o = element as PSObject;
-            if (o != null)
+            if(_rangeKind.HasValue)
             {
-                element = o.BaseObject;
+                dynamic val = element;
+                switch (_rangeKind)
+                {
+                    case ValidateRangeKind.Positive:
+                        if(val <= 0)
+                        {
+                            throw new ValidationMetadataException(
+                                "ValidateRangeTooSmall",
```

Comment:
It's convenient but not important for the ErrorId to match the name of the property in the resx file.

Note that we may have a few legacy places where we do use the ErrorId to load the resource - in those cases, it is important that they match, but it would be even better to convert those locations to this pattern where the ErrorId and format string are separate parameters.

---
In reply to: [123883907](https://github.com/PowerShell/PowerShell/pull/4084#discussion_r123883907) [](ancestors = 123883907)